mysql/index.d.ts

```diff
@@ -151,24 +151,24 @@ interface IQuery {
 
     on(ev: string, callback: (...args: any[]) => void): IQuery;
     on(ev: 'error', callback: (err: IError) => void): IQuery;
-    on(ev: 'fields', callback: (fields: any, index: number) => void): IQuery;
+    on(ev: 'fields', callback: (fields: IFieldInfo[], index: number) => void): IQuery;
     on(ev: 'result', callback: (row: any, index: number) => void): IQuery;
     on(ev: 'end', callback: () => void): IQuery;
 }
 
 interface IQueryFunction {
     (sql: string): IQuery;
-    (sql: string, callback: (err: IError, ...args: any[]) => void): IQuery;
-    (sql: string, values: Array<any>): IQuery;
-    (sql: string, values: Array<any>, callback: (err: IError, ...args: any[]) => void): IQuery;
+    (sql: string, callback: (err: IError, results?: any, fields?: IFieldInfo[]) => void): IQuery;
```

Comment:
I'll have to run some tests to see if all arguments are provided for every query type. The mysqljs documentation is a bit lacking.